### PR TITLE
fix: Some HTTP requests sent by apps don't have their data parsed into JSON

### PR DESCRIPTION
**Pull request by d-gubert**
## Proposed changes (including videos or screenshots)

Removes the validation in the `jsonParser` function in `@rocket.chat/server-fetch` package to allow the API to stringify the body of requests of any method, as long as the request has the content type "application/json"

This was the default behavior prior to adopting the package.

## Further comments

The validation checking for either POST, PUT or DELETE request methods before stringifying the body of the request broke compatibility with the Apps-Engine API. Besides that, it would make sense for _any_ request with the header `Content-Type: application/json` and a serializable object as body to have this object stringified - otherwise the request will be sent with a body of `[object Object]` - and stringifying the content of other methods (GET, HEAD, CONNECT, etc.) is not detrimental to the API in any way

### CODE REVIEW

**Review by geekgonecrazy — APPROVED** [review]
Would still like to understand why this check was even added?

**Review by murtaza98 — APPROVED** [review]
Verified that this fix works with the help of Salesforce App...where we'd initially detected this issue

**Comment by murtaza98** [comment]
Thanks @d-gubert :raised_hands:

File: packages/server-fetch/src/parsers.ts [hunk 1]
```diff
@@ -1,32 +1,20 @@
...
-function isPostOrPutOrDeleteWithBody(options?: ExtendedFetchOptions): boolean {
-	// No method === 'get'
-	if (!options?.method) {
-		return false;
-	}
-	const { method, body } = options;
-	const lowerMethod = method?.toLowerCase();
-	return ['post', 'put', 'delete'].includes(lowerMethod) && body != null;
-}
-
```
[the remark this answers is not in the document]
Comment (d-gubert) [reply packages/server-fetch/src/parsers.ts:12]: Resolving for the need to merge, but we still want your input here @KevLehman :P

**Pull request merged**